share/spice/quixey/quixey.js

```diff
@@ -244,50 +237,29 @@
         return "$" + (p/100).toFixed(2).toString();
     }
 
-    // template helper for price formatting
-    // {{price x}}
-    Handlebars.registerHelper("Quixey_qprice", function(obj) {
-        "use strict";
-
-        return qprice(obj);
-    });
-
-
-    var quixey_image = function(image_url) {
-        return "http://" + quixey_image_domain + image_url.match(/\/image\/.+/)[0];
-    }
-
-    var make_icon_url = function(item) {
-        var domain = quixey_image_domain,
-            icon_url = item.icon_url;
+    function make_icon_url(item) {
+        var icon_url = item.icon_url || null;
 
         if (!icon_url) {
-            // console.warn("quixey: icon_url is null for %o", item);
-            if (item.editions && item.editions[0].icon_url)
-                icon_url = item.editions[0].icon_url;
-            else
-                return null;
-        }
-
-        // Get the image server that the icon_url in platforms is pointing to.
-        // It's not ideal, but the link to the app's image still has to redirect
-        // and it redirects to HTTPS. What we want is an HTTP link (for speed).
-        if (item.platforms && item.platforms.length > 0 && item.platforms[0].icon_url) {
-            domain = item.platforms[0].icon_url.match(/https?:\/\/([^\/]+)/)[1];
+            if (item.editions){
+                $.each(item.editions, function(index, edition) {
+                    if (edition.icon_url){
+                        icon_url = item.editions.icon_url;
+                        return false
+                    }
+                });
```

Comment:
FYI--you can use `item.editions.forEach(function(edition, index){ ... }` now that we dropped IE8 support :)